stm32f303: Add lacking SoC-level spi dt-node description.

kuliga opened the pull request:
The stm32f303 SoC has a spi3 peripheral not listed in current dts.
Add lacking description.

Signed-off-by: Jan Kuliga <jtkuliga@gmail.com>

kuliga committed:
stm32f303: Add lacking SoC-level spi dt-node description.

The stm32f303 SoC has a spi3 peripheral not listed in current dts.
Add lacking description.

Signed-off-by: Jan Kuliga <jtkuliga@gmail.com>

erwango reviewed (changes requested):
Thanks for this addition.

This is fine but SPI3 is not available on stm32f303X8 parts.
So please add `/delete-node/ &spi3;` in stm32f303X8.dtsi.
(This should also have been done for spi2 initially.)

kuliga committed:
stm32f303x8: Exclude missing spi dt-nodes.

The stm32f303x8 SoC is not equipped with spi2 and spi3 peripherals.
Exclude them from the SoC-level devicetree.

Signed-off-by: Jan Kuliga <jtkuliga@gmail.com>

ABOSTM reviewed (approved):
LGTM
